src/js/component.js

```diff
@@ -1384,6 +1385,84 @@ class Component {
     return intervalId;
   }
 
+  /**
+   * Creates an Interval respects the visibility of the document.
+   * It will NOT run when the document is hidden, and will run at all other
+   * times.
+   *
+   * @param {Component~GenericCallback} fn
+   *        The function to run every `x` seconds, unless document is hidden
+   *
+   * @param {number} interval
+   *        Execute the specified function every `x` milliseconds unless the document
+   *        is hidden
+   *
+   * @return {number} guid
+   *         The guid of the visible interval that can be used to clear it in
+   *         {@link Component#clearVisibleInterval}
+   */
+  setVisibleInterval(fn, interval) {
+    this.visibleIntervals_ = this.visibleIntervals_ || {};
```

Comment:
open to suggestions on a better way to do this, but It seemed like the simplest solution from the user perspective.